### PR TITLE
chore: began creating the search by Ings screen

### PR DESCRIPTION
**Pull request by stefanomusicman**
# Description of the problem
began creating the search by Ingredients screen

## How does it solve the problem
different parts have been split in order to keep code lean, following same patterns and procedures as usual

# References (also the Jira ticket)
BAR-46

# Checklist:

- [x] I re-reviewed the Jira ticket now to make sure 
- [x] My code follows the style guidelines of this project & of Dart
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have wrote some documentation if I changed a big behavior of the code
- [x] My changes generate no new warnings/and if they do, I described them to the team
- [ ] I have added tests that prove my fix is effective or that my feature works if I worked on a logic class/function
- [ ] New and existing unit and integration tests pass locally with my changes

### CODE REVIEW

**Comment by stefanomusicman** [comment]
Untitled
Untitled1

**Comment by danielemonaca** [comment]
Wow, it's impressive really!

As the PR is not set in `ready for review` I didn't do a full review, but here are some quick things I found:

- The `rectangle.svg` is a duplicate of `white_bottom_part.svg` that was used in the intros, you could think about extracting that
- You access the page by clicking the `Search By Ingredients` button in the home, not the search bar
- When you click the search bar/plus icon right now it just pushes the keyboard, the requirements from when you click the bottom panel are the following:
    1. Expands the panel( not to the totally top, but like the same height on you could do it manually)
    2. After that, you can pull up the keyboard, which should not push the design as it does now, but would go on top of the design.


Let me know when you have questions

File: lib/ui/screens/home/discover_new_ones.dart [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
+import 'package:barbeat/ui/screens/home/data/get_card_info.dart';
 import 'package:flutter/material.dart';
 
 class DiscoverNewOnes extends StatelessWidget {
   const DiscoverNewOnes({super.key});
 
   @override
   Widget build(BuildContext context) {
-    final data = gridData;
+    // final data = gridData;
```
Comment (danielemonaca) [inline lib/ui/screens/home/discover_new_ones.dart:10]: Do we still need this comment?

File: lib/ui/screens/searchByIngs/panel_widget.dart [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  Widget _buildDragHandle() {
+    return Center(
+      child: Padding(
+        padding: const EdgeInsets.only(top: 8, bottom: 15),
+        child: Container(
+          width: 50,
+          height: 5,
+          decoration: BoxDecoration(
+            color: const Color(0xFFD9D9D9),
```
Comment (danielemonaca) [inline lib/ui/screens/searchByIngs/panel_widget.dart:108]: What color is that? 😆

File: lib/ui/screens/searchByIngs/searchByIngs_background.dart [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        ),
+        _buildWhiteBottomPart(size),
+      ],
+    );
+  }
+
+  Widget _buildWhiteBottomPart(Size size) {
+    const whiteBottomPart = 'assets/introImages/white_bottom_part.svg';
+    return Align(
+      // bottom: -4,
```
Comment (danielemonaca) [inline lib/ui/screens/searchByIngs/searchByIngs_background.dart:56]: Can we remove this?

File: lib/ui/screens/searchByIngs/searchByIngs_background.dart [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      // top: 5,
```
Comment (danielemonaca) [inline lib/ui/screens/searchByIngs/searchByIngs_background.dart:57]: Can we remove this?

File: lib/ui/screens/searchByIngs/searchByIngs_main.dart [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      child: Scaffold(
+        resizeToAvoidBottomInset: false,
+        backgroundColor: backgroundColor,
+        body: SlidingUpPanel(
+          onPanelClosed: () => FocusManager.instance.primaryFocus?.unfocus(),
+          controller: panelController,
+          body: Stack(
+            children: [
+              const Background(),
+              NoIngsPic(size: size), // TODO: render conditionally***
```
Comment (danielemonaca) [inline lib/ui/screens/searchByIngs/searchByIngs_main.dart:34]: Nice TODO comment here

File: pubspec.yaml [hunk 6]
```diff
@@ -58,7 +58,10 @@ dependencies:
   firebase_core: ^2.1.1
   auto_size_text: ^3.0.0
   simple_shadow: ^0.3.0
+  sliding_up_panel: ^2.0.0+1
   rive: ^0.9.0
+  flutter_keyboard_visibility: ^5.4.0
```
Comment (danielemonaca) [inline pubspec.yaml:63]: Fun fact:
I thought that adding so many packages would hurt the final app's size, but then I saw this:

> Dart comes with a Tree Shaker algorithm that will remove all unused code in release mode, reducing the size of your app to only the minimum.
> 
> So there's no real comparison with java and the jar size. Only the code you use will be added to your application.

**Comment by danielemonaca** [comment]
image
I think the text has some mismatch in terms of font weight, could you try to re-adjust it?

**Comment by danielemonaca** [comment]
image
The background is not white but is `#FAF8F2`

If you now try and fetch the latest main branch, you will have available this code here:
```
import 'package:barbeat/common_libs.dart';
import 'package:flutter_svg/flutter_svg.dart';
import 'package:simple_shadow/simple_shadow.dart';

Widget buildWhiteBottomPart(Size size, {double? height, Color? color}) {
  const whiteBottomPart = 'assets/introImages/white_bottom_part.svg';
  const shadowWhiteBottomPartColor = Color(0xFFC7C5C1);
  return Positioned(
    bottom: -40,
    child: SizedBox(
      width: size.width,
      height: size.height * (height ?? 0.65),
      child: SimpleShadow(
          opacity: 0.40,
          color: shadowWhiteBottomPartColor,
          offset: const Offset(-10, -10),
          sigma: 7,
          child: SvgPicture.asset(
            whiteBottomPart,
            fit: BoxFit.cover,
            color: color ?? Colors.white,
          )),
    ),
  );
}
```

so you can just replace your background with something like
`final Color _colorForArchBackground = const Color(0xFFFAF8F2);`
...
`buildWhiteBottomPart(size, height: 0.60, color: _colorForArchBackground)`

**Comment by danielemonaca** [comment]
I have just seen something that we can improve out of the search field, so I created a ticket for it for the next sprints: https://unboolean.atlassian.net/jira/software/projects/BAR/boards/3/backlog?selectedIssue=BAR-83

**Review by danielemonaca — CHANGES_REQUESTED** [review]
Some UI changes are needed, but functionality-wise, it's incredible!!

File: pubspec.yaml (hunk 6 above)
Comment (stefanomusicman) [reply pubspec.yaml:63]: good to know!

File: lib/ui/screens/home/discover_new_ones.dart (hunk 1 above)
Comment (stefanomusicman) [reply lib/ui/screens/home/discover_new_ones.dart]: nope, i removed it

File: analysis_options.yaml [hunk 7]
```diff
@@ -24,6 +24,7 @@ linter:
   rules:
     # avoid_print: false  # Uncomment to disable the `avoid_print` rule
     # prefer_single_quotes: true  # Uncomment to enable the `prefer_single_quotes` rule
+    file_names: false
```
Comment (danielemonaca) [inline analysis_options.yaml:27]: What is this?

**Comment by danielemonaca** [comment]
I think there is an issue with the most popular container? 
image

**Review by danielemonaca — APPROVED** [review]
Very very nice

**Pull request merged**